src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -0,0 +1,517 @@
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("min_size", (int) 0);
+  SetInputParam("epsilon", (double) 0.1);
+
+  mlpackMain();
+
+  arma::Row<size_t> output;
+  output = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_EQUAL(1, arma::accu(output < inputData.n_cols));
+}
+
+/**
+ * Check that the cluster assignment is different for different
+ * values of epsilon.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffEpsilonTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that the cluster assignment is different for different
+ * values of Min Size.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffMinSizeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+  SetInputParam("min_size", (int) 5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["min_size"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+  SetInputParam("min_size", (int) 40);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that number of noise points increases as
+ * the value of epsilon decreases.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANEpsilonNoisePointsTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  int noisePointsOutput1 = 0;
+
+  for (size_t i = 0; i < output1.n_elem; ++i)
+  {
+    if (arma::accu(output1[i] > inputData.n_cols))
```

Comment:
Why use `arma::accu()` here?  A noise assignment will be `SIZE_MAX` (I updated the documentation, which was incorrect, in d984e10e4; sorry about that), so you can just compare against that value.

Reply:
Sorry, It's on my part. I suggested Nikhil to either use `noisePointOutput2 = arma::accu(output2 > UPPER_LIMIT_OF_DATASET_POINTS)` instead of calculating using garbage value or to use some kind of numeric_limit here. I wasn't able to give enough points on part of numeric_limit.
Well, I think it would be better to use  `noisePointOutput2 = arma::accu(output2 == SIZE_MAX)` to calculate number of points.